extensions/blocks/podcast-player/edit.js

```diff
@@ -0,0 +1,125 @@
+/* eslint-disable no-unused-vars */
+
+/**
+ * External dependencies
+ */
+import { useState } from '@wordpress/element';
+import {
+	Button,
+	Disabled,
+	ExternalLink,
+	PanelBody,
+	Placeholder,
+	RangeControl,
+	TextControl,
+	Toolbar,
+	Notice,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { BlockControls, BlockIcon, InspectorControls } from '@wordpress/block-editor';
+import ServerSideRender from '@wordpress/server-side-render';
+import { isURL } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import './editor.scss';
+import { edit, queueMusic } from './icons/';
+
+const DEFAULT_MIN_ITEMS = 1;
+const DEFAULT_MAX_ITEMS = 10;
+
+const handleSSRError = () => {
+	return <p>{ __( 'Failed to load Block', 'jetpack' ) }</p>;
+};
+
+const PodcastPlayerEdit = ( { attributes, setAttributes } ) => {
+	// Block attributes.
+	const { url, itemsToShow } = attributes;
+
+	// State.
+	const [ editing, setEditing ] = useState( ! url );
+	const [ urlError, setUrlError ] = useState( '' );
+
+	const onSubmitURL = event => {
+		event.preventDefault();
+
+		if ( url ) {
+			const isValidURL = isURL( url );
```

Comment:
We should also add a test here to check against a regex for valid Anchor FM embed urls

See here for an example:

https://github.com/Automattic/jetpack/blob/f02d252966e432088b4c3bcc76c95729b5a885a6/extensions/blocks/pinterest/edit.js#L17

Reply:
Indeed. In addition, you may also want to validate the URL on the PHP side, with `\Jetpack_Gutenberg::validate_block_embed_url()`. You can check the Calendly and the Eventbrite blocks for two different approaches (domain host and regex).

Reply:
How about `https?:\/\/anchor\.fm\/s\/[\w\d]+\/podcast\/rss`?

Reply:
I am not familiar enough with Anchor.fm to be of much help here, but would it be worth to build a utility function to extract the parameters you need from the different formats you may expect one to paste into the field, and build your own URL based on that?
A bit like how it's done for Google Calendar for example:
https://github.com/Automattic/jetpack/blob/2be4affe7acd38c4a7ad61cb3f54d42873aaba7f/extensions/blocks/google-calendar/utils.js
https://github.com/Automattic/jetpack/blob/926bdd14e214d50f15bae111f581d709bab6759d/extensions/blocks/google-calendar/test/utils.js

Reply:
> We should also add a test here to check against a regex for valid Anchor FM

if I'm not wrong, this block is not limited only to the Anchor FM service, so I think it's ok going with the current URL checking.

Reply:
Correction here but this block is not related to Anchor.fm in any way other that it supports RSS Podcast URLs which Anchor generates as one of many providers. WP-hosted podcasts will work here as well so I see no need to limit domains